### PR TITLE
Total users count fix, GetUsers fix for verified field, verified t1 count for GetUserByID

### PR DESCRIPTION
**Pull request by ice-myles**
Total users count fix: taking into the account registered users. GetUsers fix for verified field. Added verified t1 count to GetUserByID function response.

### CODE REVIEW

File: users/users_get.go [hunk 1]
```diff
@@ -35,25 +35,37 @@ func (r *repository) getUserByID(ctx context.Context, id UserID) (*UserProfile,
...
+					FROM users
+					WHERE referred_by = $1 AND 
+						  username != id AND
+						  kyc_step_passed >= %[1]v AND
+						  kyc_steps_last_updated_at IS NOT NULL AND
+						  kyc_steps_created_at IS NOT NULL AND
+						  ARRAY_LENGTH(kyc_steps_last_updated_at, 1) >= %[1]v AND
+					  	  kyc_steps_last_updated_at[%[1]v - 1] IS NOT NULL AND
+					  	  ARRAY_LENGTH(kyc_steps_created_at, 1) >= %[1]v AND
+					  	  kyc_steps_created_at[%[1]v - 1] IS NOT NULL
```
Comment (ice-ares) [inline users/users_get.go:61]: @ice-cronus  I dont think this is correct. Cuz this doesn't mean the face kyc was done. Cuz now we allow social before face. Social puts kyc step passed >= 3.

File: users/users_get.go [hunk 2]
```diff
@@ -188,7 +200,14 @@ func (r *repository) GetUsers(ctx context.Context, keyword string, limit, offset
...
 			SELECT 
-			    (u.kyc_step_passed >= %[2]v AND u.quiz_completed) 				  AS verified,
+			    (u.kyc_step_passed >= %[2]v AND
+					kyc_step_passed >= %[2]v AND
+					kyc_steps_last_updated_at IS NOT NULL AND
+					kyc_steps_created_at IS NOT NULL AND
+					ARRAY_LENGTH(kyc_steps_last_updated_at, 1) >= %[2]v AND
+					kyc_steps_last_updated_at[%[2]v - 1] IS NOT NULL AND
+					ARRAY_LENGTH(kyc_steps_created_at, 1) >= %[2]v AND
+					kyc_steps_created_at[%[2]v - 1] IS NOT NULL) 				  AS verified,
```
Comment (ice-ares) [inline users/users_get.go:210]: @ice-cronus same thing here

File: users/mining_sessions_source.go [hunk 3]
```diff
@@ -29,11 +28,7 @@ func (s *miningSessionSource) Process(ctx context.Context, msg *messagebroker.Me
 		return errors.Wrapf(err, "failed to updateMiningSession for %#v", ses)
 	}
 
-	return errors.Wrapf(multierror.Append(nil,
-		errors.Wrap(s.incrementTotalActiveUsersCount(ctx, ses), "failed to incrementTotalActiveUsersCount"),
-		errors.Wrap(s.updateTotalUsersCount(ctx, &UserSnapshot{User: usr}), "failed to updateTotalUsersCount"),
-		errors.Wrap(s.updateTotalUsersPerCountryCount(ctx, &UserSnapshot{User: usr}), "failed to updateTotalUsersPerCountryCount"),
```
Comment (ice-ares) [inline users/mining_sessions_source.go:35]: U sure u dont need this?

Comment (ice-myles) [reply users/mining_sessions_source.go:35]: Yes, our changes is to have +1 total users count only for registered users.
This removed lines are to increment them only when they passed KYC and this is first session after KYC.

Comment (ice-ares) [reply users/mining_sessions_source.go:35]: But u're not increasing them anywhere else

File: users/users_get.go (hunk 1 above)
Comment (ice-myles) [reply users/users_get.go:61]: If we consider that just passed social gets verified badge now, then this and setVerified function need to be changed ofc.

File: users/mining_sessions_source.go (hunk 3 above)
Comment (ice-myles) [reply users/mining_sessions_source.go:35]: They are increased [here](https://github.com/ice-blockchain/eskimo/blob/master/users/users_snapshot.go#L27-L31).

File: users/users_get.go (hunk 1 above)
Comment (ice-myles) [reply users/users_get.go:61]: `{null, null, social, social}` he has after he passed socials only. He will not have **verified** due to  `last_updated_at[1] is NULL`.
After he passes face kyc:
`{face, face, social, social}` 
and `then last_updated_at[1] != NULL`.

File: users/mining_sessions_source.go (hunk 3 above)
Comment (ice-ares) [reply users/mining_sessions_source.go:35]: Yea, but i think those are guarded by some conditions. Check if they still apply

Comment (ice-myles) [reply users/mining_sessions_source.go:35]: We've set for them earlier for `updateTotalUsersCount`
```
-if usr.User.isFirstMiningAfterHumanVerification(r) {
+if true || usr.User.isFirstMiningAfterHumanVerification(r) {
    return r.incrementOrDecrementTotalUsers(ctx, time.Now(), true)
```
And the same for `updateTotalUsersPerCountryCount`

**Pull request merged**